### PR TITLE
Fixed non-natural spawning near player (egg, breeding, etc.)

### LINKED ISSUE #2873: Egg spawning was broken by #2863
#2863

### LINKED ISSUE #2879: Breeding has stopped working
Breeding has stopped working on my server after update few days ago.
When I feed pair of cows/sheeps/chickens it will show hearts, gives xp, but no baby animal is born.
I think it might be caused by #2863.

### PR DESCRIPTION
**Pull request by LogicParrot**
Old PR: #2876
Closes #2873
Closes #2879

### CODE REVIEW

File: src/MobSpawner.cpp [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 #include "MobSpawner.h"
 #include "Mobs/IncludeAllMonsters.h"
-
```
Comment (bearbin) [inline src/MobSpawner.cpp]: Should this line have been removed?

Comment (LogicParrot) [inline src/MobSpawner.cpp]: Fixed.

File: src/MobSpawner.cpp [hunk 2]
```diff
@@ -128,201 +128,210 @@ eMonsterType cMobSpawner::ChooseMobType(EMCSBiome a_Biome)
...
-					) &&
-					(a_RelY >= 62) &&
-					(Random.NextInt(3) != 0)
-				);
-			}
-			
-			case mtEnderman:
+		case mtEnderman:
+		{
+			if (a_RelY < 250)
```
Comment (bearbin) [inline src/MobSpawner.cpp]: could this just break here instead of nesting?

**Comment by bearbin** [comment]
This looks OK, if it's tested than I'd say to merge.

**Pull request merged**